### ISSUE TITLE
Cannot reactivate binding using CUL transport without restarting OpenHAB

### ISSUE THREAD

**Issue by cyclingengineer** [openhab/openhab1-addons#1328]
Hi Till (@dereulenspiegel),

I've got another issue with the CUL transport to do with when a binding is restarted (e.g. when you remove it from the addons directory and then add it again). It seems to get into a state where it no longer receives anything after trying to add a listener again. The only way to resolve it is to completely restart openhab (or maybe to remove then re-add the io transport? but I haven't tried that).

I believe my code is correct in the binding - it removes the listener then requests to close the device (See https://github.com/openhab/openhab/blob/master/bundles/binding/org.openhab.binding.maxcul/src/main/java/org/openhab/binding/maxcul/internal/MaxCulBinding.java line 103).

Example log from my binding:

```
2014-08-06 23:35:22.951 DEBUG o.o.b.m.internal.MaxCulBinding[:111]- De-Activating MaxCul binding
2014-08-06 23:35:22.952 DEBUG o.o.i.t.c.i.CULSerialHandlerImpl[:135]- Closing serial device /dev/cul
2014-08-06 23:35:22.953 DEBUG o.o.i.t.c.i.AbstractCULHandler$SendThread[:64]- Error while sleeping in SendThread
java.lang.InterruptedException: sleep interrupted
        at java.lang.Thread.sleep(Native Method)
        at org.openhab.io.transport.cul.internal.AbstractCULHandler$SendThread.run(AbstractCULHandler.java:62)
2014-08-06 23:35:23.179 DEBUG o.o.b.m.internal.MaxCulBinding[:115]- CUL IO should now be closed
2014-08-06 23:35:23.191 DEBUG o.o.b.m.i.MaxCulActivator[:42]- MaxCul binding has been stopped.
2014-08-06 23:35:38.009 DEBUG o.o.b.m.i.MaxCulActivator[:34]- MaxCul binding has been started.
2014-08-06 23:35:38.237 DEBUG o.o.b.m.internal.MaxCulBinding[:100]- Activating MaxCul binding
2014-08-06 23:35:38.239 DEBUG o.o.b.m.internal.MaxCulBinding[:227]- MaxCUL Reading config
2014-08-06 23:35:38.239 DEBUG o.o.b.m.internal.MaxCulBinding[:243]- Setting up device serial:/dev/cul
2014-08-06 23:35:38.240 DEBUG o.o.b.m.internal.MaxCulBinding[:262]- Opening CUL device on serial:/dev/cul
2014-08-06 23:35:38.240 DEBUG o.o.i.transport.cul.CULManager[:52]- Trying to open device serial:/dev/cul in mode MAX
2014-08-06 23:35:38.240 DEBUG o.o.i.transport.cul.CULManager[:58]- Device serial:/dev/cul is already open in mode MAX, returning already openend handler
2014-08-06 23:35:38.251 DEBUG o.o.b.m.i.MaxCulMsgHandler[:283]- Associated MaxCulBindingMessageProcessor
```

Thanks,
Paul
##

**Comment by teichsta** [comment]
@dereulenspiegel any feedback appreciated …

**Comment by dereulenspiegel** [comment]
Does this only happen with serial devices or also network devices? Probably I am missing something when disconnecting from the serial port so reconnecting becomes an issue unless openHAB is shutdown and therefore all resources are freed on an OS level.

**Comment by 9037568** [comment]
Let's start a new ticket if this is still happening with the latest code (after #4035).
